riptide-opentelemetry/src/main/java/org/zalando/riptide/opentelemetry/OpenTelemetryPlugin.java

```diff
@@ -0,0 +1,107 @@
+package org.zalando.riptide.opentelemetry;
+
+import com.google.common.collect.Multimaps;
+import io.opentelemetry.api.GlobalOpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.context.propagation.TextMapPropagator;
+import org.springframework.http.client.ClientHttpResponse;
+import org.zalando.fauxpas.ThrowingBiConsumer;
+import org.zalando.riptide.Attribute;
+import org.zalando.riptide.Plugin;
+import org.zalando.riptide.RequestArguments;
+import org.zalando.riptide.RequestExecution;
+import org.zalando.riptide.opentelemetry.span.CompositeSpanDecorator;
+import org.zalando.riptide.opentelemetry.span.ErrorSpanDecorator;
+import org.zalando.riptide.opentelemetry.span.HttpHostSpanDecorator;
+import org.zalando.riptide.opentelemetry.span.HttpMethodSpanDecorator;
+import org.zalando.riptide.opentelemetry.span.HttpPathSpanDecorator;
+import org.zalando.riptide.opentelemetry.span.HttpStatusCodeSpanDecorator;
+import org.zalando.riptide.opentelemetry.span.PeerHostSpanDecorator;
+import org.zalando.riptide.opentelemetry.span.SpanDecorator;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.BiConsumer;
+import javax.annotation.Nonnull;
+
+public class OpenTelemetryPlugin implements Plugin {
+
+    public static final Attribute<String> OPERATION_NAME = Attribute.generate();
+
+    private final Tracer tracer;
+    private final TextMapPropagator propagator;
+    private final SpanDecorator spanDecorator;
+
+    public OpenTelemetryPlugin(final Tracer tracer, SpanDecorator... decorators) {
+        this.tracer = tracer;
```

Comment:
I'm struggling a bit with this. In OpenTracing the Tracer was the center of everything (span creation, span context propagation, span exporting, ...). That's no longer the case with OpenTelemetry. Which is why you then use `GlobalOpenTelemetry` to get the Propagators, for example. In fact, Tracers are supposed to tag the source of their instrumentation. That would mean something like `this.tracer = GlobalOpenTelemetry.getTracerProvider.get("riptide-opentelemetry");`

That being the case I see a couple of options:
1. `OpenTelemetryPlugin` has an empty constructor, and another with SpanDecorators, and we get everything from the GlobalOpenTelemetry.
2. We support a builder to allow users to pass their own Tracer and propagator.

WDYT?

Reply:
I went from spring perspective which doesn't recommend static initialisation, but injection of dependencies. Still looking after some time after I wrote this I'm not sure why I went with tracer by injection, but static init of propagators. 
Maybe it worth to go with OpenTelemetry instance as constructor parameter and take tracer and propagators out of this instance? Is this approach fits to telemetry principles?

Reply:
I get the injection argument. Certainly helps with testing.
But passing an `OpenTelemetry` will not really change much, from a user perspective. It's still not the user setting the propagators or the tracer. On the other hand, it should be riptide calling `getTracer(instrumentationName, instrumentationVersion)` in order to set the two parameters.

Let's go ahead with the `OpenTelemetry` as a constructor parameter